### PR TITLE
sat-166: Add description field for shows

### PR DESCRIPTION
**Pull request by ValeriiLav**

### CODE REVIEW

File: svc/shows/repository/sql/migrations/20210705122558-add_description_field.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- +migrate Up
+ALTER TABLE shows
+    ADD COLUMN description VARCHAR DEFAULT NULL;
+-- +migrate Down
+ALTER TABLE shows DROP COLUMN category;
```
Comment (Qweder93) [inline svc/shows/repository/sql/migrations/20210705122558-add_description_field.sql:5]: should drop description

**Comment by Qweder93** [comment]
rest looks nice

File: svc/shows/repository/sql/migrations/20210705122558-add_description_field.sql (hunk 1 above)
Comment (dmitrymomot) [reply svc/shows/repository/sql/migrations/20210705122558-add_description_field.sql]: @ValeriiLav ^^^

File: svc/shows/service.go [hunk 2]
```diff
@@ -204,6 +209,10 @@ func (s *Service) AddShow(ctx context.Context, sh Show) error {
 			String: sh.Category,
 			Valid:  true,
 		},
+		Description: sql.NullString{
+			String: sh.Description,
+			Valid:  true,
```
Comment (dmitrymomot) [inline svc/shows/service.go:214]:
```suggestion
			Valid: len(sh.Description) > 0,
```

File: svc/shows/service.go [hunk 3]
```diff
@@ -221,6 +230,10 @@ func (s *Service) UpdateShow(ctx context.Context, sh Show) error {
...
```
Comment (dmitrymomot) [inline svc/shows/service.go:235]:
```suggestion
			Valid: len(sh.Description) > 0,
```

File: svc/shows/transport.go [hunk 4]
```diff
@@ -202,6 +202,7 @@ func decodeAddShowRequest(_ context.Context, r *http.Request) (interface{}, erro
 		Cover:         r.URL.Query().Get(cover),
```
Comment (dmitrymomot) [inline svc/shows/transport.go:202]: Use HTTP methods according to REST. 
- add = POST
- update whole entity = PUT
- update partially = PATCH
All these methods have a body, and payload must be taken from it

**Review by dmitrymomot — CHANGES_REQUESTED** [review]

File: svc/shows/transport.go [hunk 5]
```diff
@@ -188,92 +175,105 @@ func codeAndMessageFrom(err error) (int, interface{}) {
...
+	}
+
+	showID := chi.URLParam(r, "show_id")
+	if showID == "" {
+		return nil, fmt.Errorf("%w: missed show id", ErrInvalidParameter)
+	}
+	req.ShowID = showID
+
+	episodeID := chi.URLParam(r, "episodes_id")
+	if showID == "" {
```
Comment (ValeriiLav) [inline svc/shows/transport.go:237]: fix it

File: svc/shows/transport.go [hunk 6]
```diff
@@ -94,48 +87,41 @@ func MakeHTTPHandler(e Endpoints, log logger) http.Handler {
...
-		decodeAddEpisodeRequest,
+	r.Get("/{show_id}/episodes}", httptransport.NewServer(
+		e.GetEpisodesByShowID,
+		decodeGetEpisodesByShowIDRequest,
 		httpencoder.EncodeResponse,
 		options...,
 	).ServeHTTP)
 
-	r.Get("/episodes/{id}}", httptransport.NewServer(
+	r.Get("/{show_id}/episodes/{episodes_id}}", httptransport.NewServer(
```
Comment (ValeriiLav) [inline svc/shows/transport.go:97]: episode_id

File: svc/shows/transport.go (hunk 5 above)
Comment (ValeriiLav) [reply svc/shows/transport.go:237]: if episodeID == "" {

**Review by dmitrymomot — CHANGES_REQUESTED** [review]
pls fix syntax errors

**Review by dmitrymomot — APPROVED** [review]

**Pull request merged**